app/services/order_cycle_distributed_products.rb

```diff
@@ -13,6 +13,21 @@ def products_relation
     Spree::Product.where(id: stocked_products).group("spree_products.id")
   end
 
+  # Joins on the first product variant to allow us to filter product by taxon. This is so
+  # enterprise can display product sorted by category in a custom order on their shopfront.
+  #
+  # Caveat, the category sorting won't work properly if there are multiple variant with different
+  # category for a given product.
+  #
+  def products_taxons_relation
+    Spree::Product.where(id: stocked_products).
+      joins("LEFT JOIN (SELECT DISTINCT ON(product_id) id, product_id, primary_taxon_id
+                        FROM spree_variants) first_variant
+                        ON spree_products.id = first_variant.product_id").
+      select("spree_products.*, first_variant.primary_taxon_id").
+      group("spree_products.id, first_variant.primary_taxon_id")
+  end
```

Comment:
I think that this is the query that should ignore deleted variants. From Konrad's testing:

> Deleted variants are taken into account when defining the sorting in the shop. This can lead to incorrect results.

This is an edge case at the moment though and I'm happy to go ahead with this PR and fix it later.